Solutions/Azure Firewall/Analytic Rules/Azure Firewall - Abnormal Deny Rate for Source IP.yaml

```diff
@@ -26,42 +28,48 @@ tactics:
   - CommandAndControl
 relevantTechniques: []
 query: |
-  let LearningPeriod = 5d;
+  let LearningPeriod = 1d;
   let RunTime = 1h;
   let StartLearningPeriod = LearningPeriod + RunTime;
```

Comment:
There seems to be a fundamental flaw in the way we are querying the data here. 
StartLearningPeriod will be now() - 25h which will never be covered by the query because we have only 24h queryPeriod.
EndRunTime will be now()+23h which will not fetch anything as we are querying in future.

Ideally, learning time should be between now()-24h to now()-1h and testing time should be now()-1h to now().
Can you please ensure that we are following this time window in all the queries according to the lookback they already have in the respective query. Please let me know if more clarity is needed from my end and I can help resolve the same over a call.

Reply:
We have changed the Query period to 25h. This should cover the startlearningperiod. End run time is 1h-1h that is 0(Current Time). I hope this should fix the issue on this query. We will fix this on other queries as well.